README.md

```diff
@@ -15,3 +16,5 @@ the function and run automatically when you commit through a GitHub action.
 
 What is the worst-case big $\Theta$ complexity of your implementation? Add your
 answer, including your reasoning, to this markdown file.
+
+- When hasCycle() is called, it checks for the base cases (constant time), and if none of those apply, then for each vertex $v$ in the graph, deploy a depth first search on each edge of $v$ with $v$ as the target. The specific worst case time complexity of this would be $v$ times the worst case time complexity of dfsCycle() since we much ensure nodes have paths to themselves. Recall from my graph search program (https://github.com/COSC3020/graph-search-rmille70) that my implementation of depth first search has a worst case time complexity bounded by $\Theta(v + e)$ where $v$ is the number of vertices in a graph, and $e$ is the number of edges. Thus the specific worst case time complexity of hasCycle() is $v(v + e)$ which can be bounded by $\Theta(v^2 + ve)$
```

Comment:
Sorry, still not following you. Why are you calling `dfsCycle` for every pair of nodes?

Reply:
To detect a cycle, I look at a specific node to make sure it has no edges pointing back it itself, and then search a subgraph starting from that node's edges to see if there exists a path back to itself. Say for example we take a graph such as the one below. Performing dfsCycle(Graph, 1, 0) would only search for paths starting from Node0's edge at index 1 to Node0 and since there are none, it returns an empty path; it does this even though there exists a cycle (Node1->Node3->Node1) downstream. In order to detect the cycle connecting Node1 to itself, we must perform dfsCycle(Graph, 0, 1), which searchs for paths from Node1's 0th edge to Node1. 

![image](https://github.com/COSC3020/detecting-cycles-rmille70/assets/112435614/47cfe779-dca8-4b6c-bb3f-60efdc9f67e6)